docs/MigratingFromTuneupJs.md

```diff
@@ -0,0 +1,328 @@
+    app.mainWindow().navigationBars()[0].buttons()[1].tap();
+    app.navigationBar().buttons()[2].tap();
+    app.mainWindow().tableViews()["Payment Type Table View"].cells()["Invoice"].tap();
+    app.mainWindow().tableViews()["Invoice Payment Table"].cells().firstWithPredicate("name = 'First Name'").scrollToVisible();
+    app.mainWindow().tableViews()["Invoice Payment Table"].cells().firstWithPredicate("name = 'First Name'").typeString("Jonny");
+    app.mainWindow().tableViews()["Invoice Payment Table"].cells().firstWithPredicate("name = 'Last Name'").scrollToVisible();
+    app.mainWindow().tableViews()["Invoice Payment Table"].cells().firstWithPredicate("name = 'Last Name'").typeString("Quest");
+    app.mainWindow().tableViews()["Invoice Payment Table"].cells().firstWithPredicate("name = 'Email Address'").scrollToVisible();
+    app.mainWindow().tableViews()["Invoice Payment Table"].cells().firstWithPredicate("name = 'Email Address'").typeString("jq@palm.com");
+    app.mainWindow().navigationBars()[0].buttons()[2].tap();
+    app.mainWindow().buttons()["Receipt Sent Done Button"].tap();
+});
+```
+
+But what is it actually doing?  Here is the exact same test in Illuminator:
+
+```javascript
+var pph = appmap.apps["PayPalHere"];
+automator.createScenario('Select invoice payment method, cancel flow, do invoice payment', ['invoice', 'US', 'C142'])
+    .withStep(pph.login.withDefaultUser)
+    .withStep(pph.orderEntry.chargeAmount, {amount: 123.45})
+    .withStep(pph.paymentOptions.byInvoice)
+    .withStep(pph.invoicePayment.back)
+    .withStep(pph.orderEntry.charge)
+    .withStep(pph.paymentOptions.byInvoice)
+    .withStep(pph.invoicePayment.sendDefaultInvoiceUponReceipt, {
+            firstName: "Jonny",
+            lastName: "Quest",
+            email: "jq@palm.com"})
+    .withStep(pph.confirmation.selectNewSale)
+    .withStep(pph.orderEntry.verifyIsActive);
+```
+
+This has the benefit of being easier to read, and is built from reusable actions (with some launch-time error checks).  On the other hand, this takes a non-trivial amount of setup done in advance via Illuminator's [AppMap](AppMap.md) (see the [Practical Introduction](PracticalIntroduction.md) for a taste of how to do that).  If you just want to get your old tests up and running ASAP, a convenience function is provided for you that wraps your entire test block into an Illuminator test action.
+
+```javascript
+var ia = appmap.apps["Illuminator"];
+automator.createScenario('Select invoice payment method, cancel flow, do invoice payment', ['myDefaultTag'])
+    .withStep(ia.do.testAsAction, {test: function () {
+        mainWindow().elements()["Login Container"].textFields()["Login Email Text field"].typeString("f@ke.user");
+        //
+        //                       -- contents snipped --
+        //
+        app.mainWindow().buttons()["Receipt Sent Done Button"].tap();
+    }});
+```
+
```

Comment:
This looks great! One suggestion here would be to provide the reader with a short list of concrete "next steps" to encourage them to migrate to "The Illuminator Way". Maybe something like:

1. Review steps in the tests and extract them into an AppMap
2. Replace test code with invocations of test steps
3. Profit?

It probably breaks down into smaller steps than that. Heck, even better would be a small example showing the original tuneup code, the first conversion to Illuminator and then the remaining transformations until it finally becomes a real Illuminator test.